frontend/public/kubevirt/components/utils/withResources.jsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import * as _ from 'lodash-es';
+
+import { inject } from '../../../components/utils';
+
+import { Loader } from '../modals/loader';
+import { showErrors } from './showErrors';
+
+
+/*
+ * Firehose helper
+ */
+export class WithResources extends React.Component {
+
+  constructor (props) {
+    super(props);
+    this.state = WithResources.getDerivedStateFromProps(props);
+    WithResources.checkErrors(this.state.errors, props.dispose);
+  }
+
+  static getDerivedStateFromProps ({ resourceMap, resources, resourceToProps }) {
+    const childrenProps = {};
+    let errors = [];
+    let loaded = true;
+
+    Object.keys(resourceMap).forEach(resourceKey => {
+      const resourceConfig = resourceMap[resourceKey];
+      const resource = _.get(resources, resourceConfig.resource.kind);
+
+      if (resource) {
+        if (resource.loaded) {
+          childrenProps[resourceKey] = resource.data;
+        } else if (resourceConfig.isRequired) {
+          loaded = false;
+        }
+
+        if (resource.loadError) {
+          errors.push(resource.loadError);
+        }
+      }
+      // skip unknown resources (CRD not created in opeshift, etc..)
```

Comment:
wouldn't it be better to set the unknown resources to empty array here rather than in `resourceToProps` callback ? IMHO its safer

Reply:
We could decide if it should be array or object based on `resourceMap.<resource_name>.resource.isList`. As a consumer of `withResources` you do not have to think if its possible that such object may not exist in openshift and remember to set it in callback. Just less error prone.

Reply:
Ok I guess it makes sense to not have such hidden behaviour there. And still user can override the default with resourceToProps callback. 

I would prefer null to be the default value, but yeah we are using it for loading. So no big deal I guess.